src/sdk/object_io.js

```diff
@@ -671,43 +724,57 @@ class ObjectIO {
 
 
 /**
+ * slice_buffers_in_range
+ * It also returns missing parts in addition to found parts(chunks)
  *
  * @param {nb.Chunk[]} chunks
  * @param {number} start
  * @param {number} end
- * @return {Buffer[]}
+ * @return {Array<{ start: number, end: number, data: Buffer | null }>}
  */
 function slice_buffers_in_range(chunks, start, end) {
     if (end <= start) {
         // empty read range
         return null;
     }
     if (!chunks || !chunks.length) {
-        dbg.error('no chunks for data', range_utils.human_range({ start, end }));
-        throw new Error('no chunks for data');
+        // null data means that the part is missing.
+        return [ { start, end, data: null } ];
     }
     let pos = start;
-    /** @type {Buffer[]} */
     const buffers = [];
     for (const chunk of chunks) {
         const part = chunk.parts[0];
         let part_range = range_utils.intersection(part.start, part.end, pos, end);
-        if (!part_range) continue;
+        if (!part_range) {
+            if (end <= part.start) {
+                // null data means that the part is missing.
+                buffers.push({ start: pos, end, data: null });
+                pos = end;
+                break;
+            }
+            continue;
```

Comment:
```javascript
// in this case part.end <= pos and we silently ignore this chunk
// although it would make more sense to print a warning or assert.
// --- start ------------- pos --------- end ---
//       |                  |             |
//       |[  buf  ][  buf  ]|             |
//       |                  |             |
//       |   [   part   ]   |             |
```

Reply:
ok